frontend/packages/topology/src/components/graph-view/components/nodeContextMenu.tsx

```diff
@@ -76,17 +76,23 @@ export const noRegroupWorkloadContextMenu = (element: Node) =>
   );
 
 export const groupContextMenu = (element: Node, connectorSource?: Node) => {
-  const applicationData: TopologyApplicationObject = {
-    id: element.getId(),
-    name: element.getLabel(),
-    resources: element.getData().groupResources,
-  };
-
   const graphData = element.getGraph().getData();
-  return createMenuItems(
-    kebabOptionsToMenu(groupActions(graphData, applicationData, connectorSource)),
-  );
+  return [
+    <ActionServiceProvider key="topology" context={{ 'topology-context-actions': element }}>
+      {({ options, loaded }) =>
+        loaded &&
+        createMenuItems(kebabOptionsToMenu(groupActions(options, graphData, connectorSource)))
+      }
+    </ActionServiceProvider>,
+  ];
 };
 
-export const graphContextMenu = (graph: Graph, connectorSource?: Node) =>
-  createMenuItems(kebabOptionsToMenu(graphActions(graph.getData(), connectorSource)));
+export const graphContextMenu = (graph: Graph) => {
+  return [
+    <ActionServiceProvider key="topology" context={{ 'topology-context-actions': graph }}>
+      {({ options, loaded }) =>
+        loaded && createMenuItems(kebabOptionsToMenu(graphActions(options)))
```

Comment:
Can you try using `createContextMenuItems` ?

Reply:
Updated. Thanks!